### DISCUSSION TITLE
Session returning unwanted data from getServerSession()

### DISCUSSION THREAD

**Discussion by Nezo96** [nextauthjs/next-auth#7381] [Help] [answered]
Proble:
When getting session with `getServerSideProps` with `getServerSession` iam getting data that i did not specified in jwt and session callbacks in `[...nextauth].js`
My api call only return id, username, realname and isLogged from db.

jwt: 
```
async jwt({ token, user }) {
      if (user) {
        token.id = user.id;
        token.username = user.username;
        token.realname = user.realname;
        token.isLogged = user.isLogged;
      }
      return token;
    },
```
session:
```
async session({ session, token }) {
      if (token) {
        session.user.id = token.id;
        session.user.username = token.username;
        session.user.realname = token.realname;
        session.user.isLogged = token.isLogged;
      }
      return session;
    },
```
As you can see there is no name, email, image when i console log session i am getting  what i defined: id, username, realname, isLogged but also name, email, image.

Now i am getting error with serializing because of name which is undefined.
`Error serializing .session.user.name retur`

**Comment by Mikk36** [comment]
Check the values of token and session (console.log them, perhaps) before you start modifying them.

**Reply by Nezo96** [reply]
In docs v3 there is this: 
```
getSession()

Client Side: Yes
Server Side: Yes
```

but in v4 is this:
```
getSession()[​](https://next-auth.js.org/getting-started/client#getsession)

Client Side: Yes
Server Side: No (See: [getServerSession()](https://next-auth.js.org/configuration/nextjs#unstable_getserversession)
```
V4docs: On the server side, this is still available to use, however, we recommend using getServerSession going forward.

My version is: next-auth": "^4.22.1"
If i use getServerSession i get this error with unwanted data
![D4A9B7FB-D23E-4017-8BC2-EEF70BD190EF](https://user-images.githubusercontent.com/39273050/235055595-67f22408-8c86-494a-9e2d-8e4548fd3147.png)

But if i use getSession() there is no additional data, only that i want

![80B5B224-E2D0-4461-B503-0C219B64760D](https://user-images.githubusercontent.com/39273050/235055770-6cb2dcda-5e06-4a2e-a196-3c2a5c0da211.png)

Maybe it is bug or i messed something 👀

**Comment by dever23b** [comment, marked as answer]
Consider overwriting the token, rather than augmenting it.  

Your `jwt` callback seems to assume that the token object is empty and simply manipulates it and then returns and the original object with your manipulations.  [The code that calls the jwt callback](https://github.com/nextauthjs/next-auth/blob/f62c016725bf65e80a3cf5b6c59293120492fc4e/packages/next-auth/src/core/routes/callback.ts#L385-L399) supplies a default token object that contains the extraneous data.  

Instead of this:
```typescript
  async jwt({ token, user }) {
    if (user) {
      token.id = user.id;
      token.username = user.username;
      token.realname = user.realname;
      token.isLogged = user.isLogged;
    }
    return token;
  }
```

Try this:
```typescript
  async jwt({ token, user }) {
    if (user) {
      token = {
        id: user.id,
        username: user.username,
        realname: user.realname,
        isLogged: user.isLogged
      }

      return token
    }
  }
```
